components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -846,7 +846,15 @@ protected void findPreviousAuthenticatedSession(HttpServletRequest request, Auth
         }
         // Get service provider chain
         SequenceConfig effectiveSequence = getSequenceConfig(context, request.getParameterMap());
-        setStepConfigAuthenticatorList(effectiveSequence);
+        String applicationName = effectiveSequence.getApplicationConfig().getApplicationName();
+        // organization SSO IDP is added for portal apps only if requested with FIDP param.
+        if (FrameworkConstants.Application.CONSOLE_APP.equals(applicationName) ||
+                FrameworkConstants.Application.MY_ACCOUNT_APP.equals(applicationName)) {
+            String[] fidpParam = request.getParameterMap().get(FrameworkConstants.RequestParams.FEDERATED_IDP);
+            if (fidpParam == null || !ORGANIZATION_LOGIN_HOME_REALM_IDENTIFIER.equals(fidpParam[0])) {
```

Comment:
Shall we validate the array length to avoid IndexOutOfBoundsException

Reply:
Fixed by - https://github.com/wso2/carbon-identity-framework/pull/5665/commits/6aad6fb8a4651e8e7aa8339f871a729a8e68d3b6